### PR TITLE
transactions by range management

### PR DESCRIPTION
**Pull request by IViol**
- flow types for config.js
- fix BigNumber types
- fix transactions types
- implement storing of transactions by block number (`toBlock` number)
- fix scheduler types
- update displaying of ETH transactions with deploying of contract
- add sorting (by block timestamp) of transactions in UI
- add initialising of transactions with empty structures at start of syncing
- update fetching of transactions
- implement re-syncing of transactions if some requests was failed or not finished

### CODE REVIEW

File: src/utils/numbers/isZero.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// @flow
+
+import { BigNumber } from 'bignumber.js'
+
+function isZero(amount: string): boolean {
```
Comment (flexxnn) [inline src/utils/numbers/isZero.js]: Here, we can use `string | number | BigNumber`

File: src/utils/type/index.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 // @flow
 
+import isVoid from './isVoid'
 import isArray from './isArray'
 import isObject from './isObject'
 import isString from './isString'
 import isNumber from './isNumber'
```
Comment (stcherenkov) [inline src/utils/type/index.js]: Lodash has all of these, btw

File: src/utils/numbers/isZero.js (hunk 1 above)
Comment (IViol) [reply src/utils/numbers/isZero.js]: fixed [here](https://github.com/jibrelnetwork/jwallet-web/pull/441/commits/8ba1ac5589d08555665366aaa738b04ec34e6a2b)

File: src/utils/type/isVoid.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// @flow
+
+function isVoid(data: any): boolean {
+  return (data == null)
```
Comment (Creepypoke) [inline src/utils/type/isVoid.js]: Why are you using round brackets?
Using the abstract equality (`==`) operator is deliberate?

Comment (IViol) [reply src/utils/type/isVoid.js]: (data == null) checks null or undefined

File: src/utils/transactions/getAddressLink.js [hunk 4]
```diff
@@ -2,8 +2,10 @@
 
 import config from 'config'
 
-function getAddressLink(address: Address, blockExplorerSubdomain: string): string {
-  return `//${blockExplorerSubdomain}.${config.blockExplorerLink}/address/${address}`
+function getAddressLink(address: ?Address, blockExplorerSubdomain: string): string {
+  return address
+    ? `//${blockExplorerSubdomain}.${config.blockExplorerLink}/address/${address}`
+    : ''
```
Comment (Creepypoke) [inline src/utils/transactions/getAddressLink.js]: I think, we can get strict `Address` type as parameter without nullable state, and check nullable state of parameter before call this function.

File: src/utils/digitalAssets/parseBalance.js [hunk 5]
```diff
@@ -1,16 +1,9 @@
...
 import { divDecimals } from '../numbers'
 
-function parseBalance(
-  value: string,
-  decimals: number = 18
-): string {
-  const normalized = divDecimals(new BigNumber(value), decimals)
-
-  return normalized.toString()
+function parseBalance(value: string, decimals: number = 18): string {
```
Comment (Creepypoke) [inline src/utils/digitalAssets/parseBalance.js]: What does number 18 mean? Maybe, we can set constants for "Magical numbers"?

**Pull request merged**